ovirt_imageio/client/_upload.py

```diff
@@ -0,0 +1,169 @@
+        default="qcow2",
+        help="Upload image format (default qcow2).")
+
+    cmd.add_argument(
+        "--sparse",
+        action="store_true",
+        help="Create sparse disk. Cannot be used with raw format on "
+             "block storage.")
+
+    cmd.add_argument(
+        "--id", "--disk-id",
+        type=_options.UUID,
+        help="A UUID for the new disk. If not specified oVirt will "
+             "create a new UUID.")
+
+    cmd.add_argument(
+        "--disable-backup",
+        dest="enable_backup",
+        action="store_false",
+        help="Creates a disk that can not be used for incremental backup.")
+
+    cmd.add_argument(
+        "filename",
+        type=_options.File,
+        help="Path to image (e.g. /path/to/image.raw). "
+             "Supported formats: raw, qcow2, iso.")
+
+
+def upload_disk(args):
+    with _ui.ProgressBar() as pb:
+        disk_info = _create_disk_info(args.filename, args.format)
+        con = _ovirt.connect(args)
+        with closing(con):
+            pb.phase = "creating disk"
+            disk = _ovirt.add_disk(
+                con=con,
+                name=disk_info.name,
+                initial_size=disk_info.initial_size,
+                provisioned_size=disk_info.provisioned_size,
+                sd_name=args.storage_domain,
+                id=args.id,
+                sparse=args.sparse,
+                enable_backup=args.enable_backup,
+                content_type=disk_info.content_type,
+                format=disk_info.format)
+
+            pb.phase = "creating transfer"
+            host = _ovirt.find_host(con, args.storage_domain)
+            transfer = _ovirt.create_transfer(con, disk, host=host)
+            try:
+                pb.phase = "uploading image"
+                _api.upload(
+                    args.filename,
+                    transfer.transfer_url,
+                    args.cafile,
+                    buffer_size=args.buffer_size,
+                    progress=pb,
+                    proxy_url=transfer.proxy_url,
+                    max_workers=args.max_workers)
+            except Exception:
+                _ovirt.cancel_transfer(con, transfer)
+                raise
+            pb.phase = "finalizing transfer"
+            _ovirt.finalize_transfer(con, transfer, disk)
+        pb.phase = "upload completed"
+
+
+def _create_disk_info(filename, dst_format):
+    # Obtain the image info dictionary.
+    img_info = _api.info(filename)
+    if img_info["format"] not in ("qcow2", "raw"):
+        raise RuntimeError(f"Unsupported image format {img_info['format']}")
+    # Obtain all reimaining fields for DiskInfo.
+    format = dst_format or img_info["format"]
+    initial_size = _api.measure(filename, format)["required"]
+    content_type = _detect_disk_content_type(
+        img_info["filename"], img_info["format"])
+    name = _get_disk_name(
+        img_info["filename"], img_info["format"], content_type)
```

Comment:
Since we don't need the extension, we can use:

    name = os.path.splitext(os.path.basename(image_info["filename"]))[0]